fuzz/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM amazonlinux:2023
+FROM amazonlinux:2023@sha256:521e488723a1ae16055067e3b78954870cf1c08c9d45421b9f8ffad82b349e1c
```

Comment:
Also interetsing choice of an image. Whats the reason we use it @sepehrdaddev ?

Reply:
> It is not clear why we need to pin as we do not deploy this image.
> 
> If we pin then we'd need to add an entry like https://github.com/zalando/skipper/blob/2fd26a2d1784a6942f64d33d463f7366ae848e21/.github/dependabot.yml#L17C1-L20

please go to "security" tab and find the docs yourself from the scorecard OpenSSF scanner.

Reply:
I read it and it only relevant for deploy-able artifacts. We don't deploy this image, its just a handy wrapper to avoid installation of go dependencies on the fuzzing machine

Reply:
@AlexanderYastrebov, I used it because it was a bit easier, with Alpine due to musl libc it didn't work (the fuzzer crashes)  and with debian, the clang versions were ancient.